src/commands/xkcd.ts

```diff
@@ -61,17 +61,20 @@ async function _getComic(endpoint: string | number): Promise<GetComicResponse> {
 	}
 }
 
+const builder = new SlashCommandBuilder()
+	.setName('xkcd')
+	.setDescription('Fetches the most recent xkcd comic, or a selected one.')
+	.setDMPermission(true)
```

Comment:
I wonder what would happen if `setDMPermission` receives `true`, but `requiresGuild` is also set to `true`. Before, TypeScript would prevent that from happening, and I'd even been planning to refactor to make `requiresGuild` obsolete by using the `dmPermission` field instead, but with that value now in the builder, I'm not sure how to read it statically and send the right context without keeping the `requiresGuild` field.

Reply:
Hmm, I did check and see that the DM permission is true by default for global commands, so I'm probably going to remove that. I'll see what happens if a guild command has the dm permission set to true and if there is a good static way to catch that. I think I have an idea.

Reply:
@AverageHelper Is your idea to be that commands with the dm permission are global and ones without are guild commands? I think it might be better to keep the types separate and find a way to just deny guild commands that try to set the dm permission to true.

Also I tested it, if you do `setDMPermission(true)` on a guild command, it doesn't complain but you can't use it in a DM. So imo it would be a non-issue, but I can see if I can find a way to make a unit test for it so that such commands can't be merged in regardless.

Reply:
The idea is that most commands that require guild information will never have a good fallback for when they don't have that information (fetching guild member stuff, channels, etc.) so I intended for the two worlds to receive a different Context type. Commands that require guild stuff should only be accessible there, so the context can have `guild`, `member`, etc. populated with something non-null always.

Ideally, there'd be a way to tell TypeScript and Discord that certain commands require guild info, and have that context statically known so that command implementers don't have to think about handling edge cases that'll never be.

Reply:
My solution though would probably be to keep the `requiresGuild `boolean around, but it really just determines if a command is a guild command or a global command. Maybe it should be renamed to `isGuildCommand`. Then the dmPermission can be set on global commands still.

My worry for using the dmPermission field for determining if a command is a global or guild command is it prevents global commands from being not allowed in DMs. Also it will be nice to control what is a guild command and what is a global command, cause of the limits imposed on the amount of both.

Reply:
Also I was able to write a unit test that prevents a command with requiresGuild: true from having the dm permission set to true, if that seems like a good part of the solution.

Reply:
That would be helpful.

What if we renamed `requiresGuild` to `dmPermission`, inverted its value in the existing command definitions (flip `true` to `false` where its used, and vice versa), and used that value to call `setDMPermission` on the builder when we deploy?

Reply:
That way there'd be less code duplication. Could still use `setDMPermission` in the builder, but it wouldn't work since the one on the `GuildCommand` or `GlobalCommand` object would take precedence

Reply:
I'm okay pushing this down the road. In the meantime, we can manually check in reviews that command authors aren't accidentally setting `setDMPermission` to the same value as `requiresGuild`.